Fix: Creating edge-commit to update the ostree repo will now refer to the existing ostree commit rather than building a independent commit

amepatil-git opened the pull request:
# Description

<!--
Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.
List any dependencies that are required for this change.
-->

Whenever we initiate a edge-commit to update existing ostree repo, its not referring to the existing ostree repo as a parent commit but instead builds an independent commit.
Due to this when we update rpm-ostree system using this repo, problems have been observed with regards to UID of the system users which are added by the rpm scriptlet since these are stored in /usr/lib/passwd which changes across the commit since the commit are now not related to each other.

This change aims to solve this problem by the follow changes,

 * Change the "if" condition in start_compose.py so that the parameters "ostree_url" and "ostree_ref" are not ignored for edge-commit

 * Added parameters "ostree_url" and "ostree_ref" to the builder role "Start compose" task such that they can be used in when updating existing commit and can be ignored when building the first commit

FIXES: <!-- AAP-NNNN  --> #477 


## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

## Checklist

- [ ] Added changelog fragment
- [ ] Tests exist for affected features covering positive and negative scenarios

amepatil-git committed:
Whenever we initiate a edge-commit to update existing ostree repo, its
not referring to the existing ostree repo as a parent commit but instead
builds an independent commit.
Due to this when we update rpm-ostree system using this repo, problems
have been observed with regards to UID of the system users which are
added by the rpm scriptlet since these are stored in /usr/lib/passwd
which changes across the commit since the commit are now not related to
each other.

This change aims to solve this problem by the follow changes,

 * Change the "if" condition in start_compose.py so that the parameters
   "ostree_url" and "ostree_ref" are not ignored for edge-commit

 * Added parameters "ostree_url" and "ostree_ref" to the builder role
   "Start compose" task such that they can be used in when updating
   existing commit and can be ignored when building the first commit

amepatil-git commented:
I have removed the when conditional being run for the block for when `builder_ostree_url is not defined` so that it runs all task regardless of whether this is a first commit or update to the existing ostree repo.

Then I have added `ostree_url` parameter with the `default(omit)` filter such that during first commit when there is no existing ostree repo `builder_ostree_url` is not defined hence `ostree_url`  will be ignored.
While when updating the commit when we pass the the location of the existing ostree repo `-e builder_ostree_url=http://X.X.X.X/test_blueprint/repo/` it able to build on top of the existing repo rather than building an independent commit.
~~~
        - name: Start compose
          infra.osbuild.start_compose:  # noqa only-builtins
            blueprint: "{{ builder_blueprint_name }}"
            compose_type: "{{ _builder_compose_type }}"
            ostree_url: "{{ builder_ostree_url | default(omit) }}"  # noqa yaml[line-length]
            ostree_ref: "{{ builder_blueprint_ref | default(omit) }}"
            timeout: "{{ builder_request_timeout }}"
          register: builder_compose_start_out
~~~

Please let me know your though and suggestion.

ericzolf commented on roles/builder/tasks/main.yml:
```diff
@@ -144,6 +143,8 @@
           infra.osbuild.start_compose:  # noqa only-builtins
             blueprint: "{{ builder_blueprint_name }}"
             compose_type: "{{ _builder_compose_type }}"
+            ostree_url: "{{ builder_ostree_url | default(omit) }}"  # noqa yaml[line-length]
```
I don't understand why you need a noqa here? The line is even shorter than the following line.

amepatil-git replied:
Hi Eric,

Thanks for correcting me.
While testing the `ostree_url` line with different combination I had taken the reference for this line form https://github.com/redhat-cop/infra.osbuild/blob/main/roles/builder/tasks/main.yml#L307.
And I forgot removing the comment when I changed the line in my tests.
I have now removed the comment in the new commit.

Please let me know your though and suggestion.

ericzolf reviewed (dismissed):
You need to add a changelog fragment with bugfixes level.
I'm not the expert but there is: https://ansible.readthedocs.io/projects/antsibull-changelog/changelogs/

And you also might want to add the closes #477 message to the commit message for later reference.

amepatil-git committed:
Correction to the previous commit as per PR comments.
Removing the "# noqa yaml[line-length]" since the line is shorter and
does not need this.

ericzolf commented:
@amepatil-git sorry to bug you but you seem to have overseen my comment about a changelog fragment.

amepatil-git committed:
Changes to the previous commit as per PR comments.
Added ansible change log fragment.
Closes #477

amepatil-git commented:
Hi @ericzolf ,

Sorry for the delay and for missing the ansible changelog suggestion.
I have resubmitted the commit with the ansible changelog fragment.
I am new to git, but since the upstream main branch was ahead since when I forked the repository, I have rebased main branch so that my commits are applied after the changes done in the main branch so that the history is cleaner.
Due to the rebase I needed to force push my earlier commits.

Please let me know your though and suggestion.

ericzolf committed:
Update changelogs/fragments/313-edge-commit-should-update-existing-ostee-repo.yaml

Single typo...

ericzolf reviewed (approved):
Changes look sound, I assume most pipelines are broken and would need proper review and refresh.
